### PR TITLE
fix(incremental sync): fix cache_key handling for select_by_cache_key

### PR DESCRIPTION
**Pull request by chobits**
### Summary

This is part 2 of cache_key bugfix.

1. fix(dao): don't generate ws_id for cache_key if schema is not workspaceable
2. only insert one cache_key into lmdb, and its unique key's ws_id could be a placeholder, here it uses default ws id. Because the ws_id should be handled by `:cache_key()`. And returned value of `:cache_key()` is already unique globally. An example is that `kong.cache:get()` will directly use this value as global unique key.
   * the final key-value in lmdb for cache_key will be:
   * key: `<entity_name>:<default_ws_id>:cache_key:<returned value from :cache_key()>`
   * value: `<index: item's key in lmdb>`

### Checklist

- [ ] The Pull Request has tests
- [x] A changelog file has been added to `CHANGELOG/unreleased/kong` or adding `skip-changelog` label on PR if unnecessary. [README.md](https://github.com/Kong/kong/blob/master/CHANGELOG/README.md)
- [ ] The Pull Request has backports to all the versions it needs to cover
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix [KAG-5661](https://konghq.atlassian.net/browse/KAG-5661)

[KAG-5661]: https://konghq.atlassian.net/browse/KAG-5661?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: kong/db/declarative/import.lua [hunk 1]
```diff
@@ -314,13 +314,10 @@ local function _set_entity_for_txn(t, entity_name, item, options, is_delete)
...
-    for _, wid in ipairs {ws_id, GLOBAL_WORKSPACE_TAG} do
-      local key = unique_field_key(entity_name, wid, "cache_key", cache_key)
-
-      -- store item_key or nil into lmdb
-      t:set(key, idx_value)
-    end
+    -- ws_id is a placeholder here, because cache_key is unique globally.
+    local _ws_id = get_default_workspace() -- avoid overwriting the outer ws_id
+    local key = unique_field_key(entity_name, _ws_id, "cache_key", cache_key)
+    t:set(key, idx_value)
```
Comment (chronolaw) [inline kong/db/declarative/import.lua:320]: Should we keep this comment `-- store item_key or nil into lmdb`?

File: kong/db/declarative/import.lua [hunk 2]
```diff
@@ -314,13 +314,10 @@ local function _set_entity_for_txn(t, entity_name, item, options, is_delete)
...
-
-    for _, wid in ipairs {ws_id, GLOBAL_WORKSPACE_TAG} do
-      local key = unique_field_key(entity_name, wid, "cache_key", cache_key)
-
-      -- store item_key or nil into lmdb
-      t:set(key, idx_value)
-    end
+    -- ws_id is a placeholder here, because cache_key is unique globally.
+    local _ws_id = get_default_workspace() -- avoid overwriting the outer ws_id
+    local key = unique_field_key(entity_name, _ws_id, "cache_key", cache_key)
```
Comment (chronolaw) [inline kong/db/declarative/import.lua:319]: How about
```
local key = unique_field_key(entity_name, get_default_workspace(), "cache_key", cache_key)
```

File: kong/db/strategies/off/init.lua [hunk 3]
```diff
@@ -208,18 +208,26 @@ local function select_by_field(self, field, value, options)
...
-  local unique_across_ws = schema.fields[field].unique_across_ws
-  -- only accept global query by field if field is unique across workspaces
-  assert(not options or options.workspace ~= null or unique_across_ws)
+  local ws_id
 
-  if unique_across_ws then
+  if field == "cache_key" then
+    -- align with cache_key insertion logic in _set_entity_for_txn
     ws_id = get_default_workspace()
+    -- value = the value generated by `:cache_key()`
```
Comment (chronolaw) [inline kong/db/strategies/off/init.lua:216]: Remove this line?

File: kong/db/declarative/import.lua (hunk 2 above)
Comment (chobits) [reply kong/db/declarative/import.lua]: done

File: kong/db/strategies/off/init.lua (hunk 3 above)
Comment (chobits) [reply kong/db/strategies/off/init.lua]: done

**Review by chronolaw — APPROVED** [review]

**Pull request merged**